### PR TITLE
Added linters

### LINKED ISSUE #1179: Enhance golangci-lint configuration
When reviewing https://github.com/open-telemetry/opentelemetry-operator/pull/1177 I spotted that the current [.golangci.yaml](https://github.com/open-telemetry/opentelemetry-operator/blob/bc32d5175e4e4aa1b29e2ee9ff15edcc7a782f82/.golangci.yaml) is very minimalistic.

I would consider enabling the following linters:

- [gosimple](https://github.com/dominikh/go-tools/tree/master/simple)  -it is enabled by default
- [staticcheck](https://staticcheck.io/) - enabled by default
- [ineffassign](https://github.com/gordonklaus/ineffassign) - enabled by default
- typecheck - enabled by default - it makes the errors better if the code does not compile 😉 
- [unused](https://github.com/dominikh/go-tools/tree/master/unused) - enabled by default
- [unparam](https://github.com/mvdan/unparam)
- [nolintlint](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/nolintlint/README.md)

I also like these but they may require some changes in the codebase

- [revive](https://github.com/mgechev/revive) - golint replacement
- [errcheck](https://github.com/kisielk/errcheck) - enabled by default

Might be nice to consider enabling this one as well

- [gofumpt](https://github.com/mvdan/gofumpt)

### PR DESCRIPTION
**Pull request by yuriolisa**
Resolves #1179 
Added some linters. 
In the case of Revive, I made a lot of code changes related to it, but I've not finished avoiding getting a huge PR. Therefore, I'll split this PR in 2 to enable revive as expected.

### CODE REVIEW

File: pkg/sidecar/podmutator.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ type sidecarPodMutator struct {
 
 var _ webhookhandler.PodMutator = (*sidecarPodMutator)(nil)
 
-func NewMutator(logger logr.Logger, config config.Config, client client.Client) *sidecarPodMutator {
+func NewMutator(logger logr.Logger, config config.Config, client client.Client) *sidecarPodMutator { //nolint:revive
```
Comment (secustor) [inline pkg/sidecar/podmutator.go:48]:
```suggestion
func NewMutator(logger logr.Logger, config config.Config, client client.Client) *sidecarPodMutator {
```

Nitpick: this should be added in the same PR with revive

File: .golangci.yaml [hunk 2]
```diff
@@ -24,3 +24,10 @@ linters:
     - exhaustive
     - godot
     - unparam
+    - gosimple
+    - unused
+    - staticcheck
+    - ineffassign
+    - typecheck
+    - unparam
+  #  - revive.
```
Comment (secustor) [inline .golangci.yaml:33]:
```suggestion
  #  - revive.
```

**Review by secustor — COMMENTED** [review]
Small nitpick else LGTM

File: pkg/instrumentation/podmutator.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ type languageInstrumentations struct {
 
 var _ webhookhandler.PodMutator = (*instPodMutator)(nil)
 
-func NewMutator(logger logr.Logger, client client.Client) *instPodMutator {
+func NewMutator(logger logr.Logger, client client.Client) *instPodMutator { //nolint:revive
```
Comment (secustor) [inline pkg/instrumentation/podmutator.go:52]:
```suggestion
func NewMutator(logger logr.Logger, client client.Client) *instPodMutator {
```

**Review by secustor — APPROVED** [review]

**Comment by yuriolisa** [comment]
@pavolloffay, @VineethReddy02, could you please have a look?

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**